src/service/asgi/responses.py

```
@@ -0,0 +1,115 @@
+from starlette.requests import ClientDisconnect
+from starlette.responses import ContentStream
+from starlette.types import Message, Receive, Scope, Send
+
+
+@contextlib.asynccontextmanager
+async def _collapsing_task_group() -> AsyncIterator[anyio.abc.TaskGroup]:
+    """Task group that re-raises a lone child exception instead of a group."""
+    try:
+        async with anyio.create_task_group() as task_group:
+            yield task_group
+    except BaseExceptionGroup as group:
```

Ruff (0.16.1)

[error] Undefined name `BaseExceptionGroup`. Consider specifying `requires-python = ">= 3.11"` or `tool.ruff.target-version = "py311"` in your `pyproject.toml` file.

(F821)